src/ripple/app/consensus/RCLConsensus.h

```diff
@@ -283,6 +284,23 @@ class RCLConsensus
             timerDelay_ = td;
         }
 
+        /** Destroy Object On Job Queue
+         *
+         * For an object that is time-consuming to destroy, this is an easy
+         * way to ensure that it's destroyed outside of a lock or other
+         * performance critical code path.
+         *
+         * @tparam T Any type.
+         * @param garbage Object to be destroyed.
+         */
+        template <class T>
+        void
+        dispose(T&& garbage)
```

Comment:
I like having this be generic in case it's useful elsewhere and we want to use it, or move this function more centrally. So I used your static_assert, which works just fine.